### ISSUE TITLE
Loss nan problem

### ISSUE THREAD

**Issue by guijuzhejiang** [deepinsight/insightface#1555]
I used arcface_torch to train with my own dataset as follow:
CUDA_VISIBLE_DEVICES=0,1,2,3 nohup python -m torch.distributed.launch --nproc_per_node=4 --nnodes=1 --node_rank=0 --master_addr="127.0.0.1" --master_port=1234 train.py --network r50 --loss arcface
But it encountered the loss NAN problem as follow:
Training: 2021-06-01 16:22:55,751-Speed 219.17 samples/sec   Loss 2.3655   Epoch: 0   Global Step: 14550   Required: 102 hours
Training: 2021-06-01 16:23:39,606-Speed 218.91 samples/sec   Loss 2.4382   Epoch: 0   Global Step: 14600   Required: 102 hours
Training: 2021-06-01 16:24:23,457-Speed 218.93 samples/sec   Loss 2.4757   Epoch: 0   Global Step: 14650   Required: 102 hours
Training: 2021-06-01 16:25:07,307-Speed 218.93 samples/sec   Loss 2.4868   Epoch: 0   Global Step: 14700   Required: 102 hours
Training: 2021-06-01 16:25:51,183-Speed 218.82 samples/sec   Loss nan   Epoch: 0   Global Step: 14750   Required: 102 hours
Training: 2021-06-01 16:26:33,973-Speed 224.36 samples/sec   Loss nan   Epoch: 0   Global Step: 14800   Required: 102 hours
Training: 2021-06-01 16:27:16,771-Speed 224.32 samples/sec   Loss nan   Epoch: 0   Global Step: 14850   Required: 102 hours
Training: 2021-06-01 16:27:59,556-Speed 224.38 samples/sec   Loss nan   Epoch: 0   Global Step: 14900   Required: 102 hours
Training: 2021-06-01 16:28:42,450-Speed 223.81 samples/sec   Loss nan   Epoch: 0   Global Step: 14950   Required: 102 hours

Someone said that reducing LR can avoid this mistake.
but even I reduced the learning rate to 0.001, it still happens.

**Comment by guijuzhejiang** [comment]
I use 4 V100 GPU to train. 
Batch size :48.
Input size: 300*300

**Comment by guijuzhejiang** [comment]
I tried to change s from 64 to 32, but loss would still be Nan. Is it arcface loss has a bug?

**Comment by guijuzhejiang** [comment]
Should I need to pre-train with softmax？But there is no softmax loss function in pytorch code

**Comment by lixiangMindSpore** [comment]
> Should I need to pre-train with softmax？But there is no softmax loss function in pytorch code
Your loss function has been modified incorrectly.

**Comment by JasperRice** [comment]
@guijuzhejiang In the arcface loss, it would calculate acos.
In the parameter, when there's a -1.0, it would cause NaN problem.
But no idea how to solve this.
